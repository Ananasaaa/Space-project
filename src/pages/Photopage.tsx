import { mockRoverPhoto } from '../mocks/mockRoverPhoto';
import '../styles/photopage.scss';

const backgroundStyle = {
  backgroundImage: `url(${process.env.PUBLIC_URL}/img/bg_photopage.jpg)`,
  backgroundSize: 'cover',
  backgroundPosition: 'center',
  backgroundRepeat: 'no-repeat',
};

const Photopage = () => {
  return (
    <div className="photopage" style={backgroundStyle}>
      {mockRoverPhoto.photos.map((photo) => (
        <div className="photo-container" key={photo.id}>
          <h3>{photo.camera.full_name}</h3>
          <p>Sol: {photo.sol}</p>
          <p>Date on Earth: {photo.earth_date}</p>
          <img src={photo.img_src} alt={photo.camera.full_name} />
        </div>
      ))}
    </div>
  );
};

export default Photopage;
