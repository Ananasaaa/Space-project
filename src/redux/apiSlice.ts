import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

export const nasaApi = createApi({
  reducerPath: 'nasaApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'https://api.nasa.gov/',
  }),
  endpoints: (builder) => ({
    getSpaceImage: builder.query({
      query: () => ({
        url: 'planetary/apod',
        params: {
          api_key: process.env.REACT_APP_NASA_KEY as string,
        },
      }),
    }),
    getMarsPhotos: builder.query({
      query: ({ sol = 4000, camera, page = 1 }) => ({
        url: 'mars-photos/api/v1/rovers/curiosity/photos',
        params: {
          sol,
          page,
          ...(camera ? { camera } : {}),
          api_key: process.env.REACT_APP_NASA_KEY as string,
        },
      }),
    }),
  }),
});

export const { useGetSpaceImageQuery, useGetMarsPhotosQuery } = nasaApi;
