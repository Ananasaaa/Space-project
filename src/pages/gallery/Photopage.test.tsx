import { render } from '@testing-library/react';
import Photopage from './index';
import { useGetMarsPhotosQuery } from '../../redux/apiSlice';

jest.mock('../../redux/apiSlice');

describe('Photopage component', () => {
  it('should render photos correctly', () => {
    (useGetMarsPhotosQuery as jest.Mock).mockReturnValue({
      data: {
        photos: [
          {
            id: '1',
            camera: { full_name: 'Test Camera' },
            sol: 1000,
            earth_date: '2024-04-20',
            img_src: 'https://example.com/photo.jpg',
          },
        ],
      },
      isLoading: false,
      error: null,
    });

    const { asFragment } = render(<Photopage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
