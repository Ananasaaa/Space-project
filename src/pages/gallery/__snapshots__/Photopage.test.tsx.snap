// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photopage component should render photos correctly 1`] = `
<DocumentFragment>
  <div
    class="photopage"
    style="background-image: url(/img/bg_photopage.jpg); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <div
      class="sol-block"
    >
      <form
        class="sol-form"
      >
        <label
          for="sol"
        >
          Enter Sol (day on Mars):
        </label>
        <input
          id="sol"
          type="text"
          value="1000"
        />
      </form>
    </div>
    <div
      class="photo-grid fade-in"
    >
      <div
        class="photo-container"
      >
        <div
          class="mini-loader"
        />
        <img
          alt="Test Camera"
          loading="lazy"
          src="https://example.com/photo.jpg"
          style="opacity: 0; transition: opacity 0.5s ease-in-out; width: 100%; height: auto; display: block;"
        />
        <h3>
          Test Camera
        </h3>
        <p>
          Sol: 1000
        </p>
        <p>
          Date on Earth: 2024-04-20
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
